apps/src/blockly/addons/cdoFieldButton.js

```diff
@@ -1,22 +1,35 @@
 import GoogleBlockly from 'blockly/core';
 
-/**
- * This is a customized field which the user clicks to select an option from a customized picker,
- * for example, the location of a sprite from a grid or a sound file from a customized modal.
- * @param value Optional. The initial value of the field.
- * @param validator Optional. A function that is called to validate changes to the field's value.
- * Takes in a value & returns a validated value, or null to abort a change
- * @param onClick The function that handles the field's editor.
- * @param transformText Optional. The function that handles how the field text is displayed.
- * @param icon Optional. SVG <tspan> element - if the field displays a button, this is the icon that is displayed on the button.
- */
 export default class CdoFieldButton extends GoogleBlockly.Field {
-  constructor({value, validator, onClick, transformText, icon}) {
+  /**
+   * This is a customized field which the user clicks to select an option from a customized picker,
+   * for example, the location of a sprite from a grid or a sound file from a customized modal.
+   * @param {Object} options - The options for constructing the class.
+   * @param options.value Optional. The initial value of the field.
```

Comment:
nit: Can you specify the type here?

Reply:
It can actually be any type depending on how the class is used, so I wasn't how to notate that!

Reply:
Looks like I might be able to just use `{*}`! Thanks, TIL.